src/miner.cpp

```diff
@@ -342,6 +342,8 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
         txNew.vout.resize(1);
         txNew.vout[0].scriptPubKey = scriptPubKeyIn;
         txNew.vout[0].nValue = GetBlockSubsidy(nHeight, chainparams.GetConsensus());
+        // Set to 0 so expiry height does not apply to coinbase txs
+        txNew.nExpiryHeight = 0;
```

Comment:
Hmm... if we do this, do we need the `IsCoinBase()` consensus check? I'm trying to imagine if there's any reason why we'd want a coinbase transaction to be able to expire. If we don't want them to expire, then we might separately want to enforce their expiry heights to be zero, rather than leaving the field unconstrained.

Reply:
The coinbase tx is only valid for the block it's mined in, so whether it expires or not shouldn't matter, correct? 

Would prefer to keep the `IsCoinBase()` check in `IsExpiredTx()`

Reply:
Right; I was just wondering whether we should add a consensus rule requiring `nExpiryHeight == 0` for coinbase transactions. Probably doesn't matter...

Reply:
As long as the rule is clearly specified, it doesn't matter. As I understand it, the implemented rule is that ``nExpiryHeight`` is ignored for coinbase transactions. However, ZIP 203 does not specify that. Please make the ZIP and implementation consistent.

Reply:
Probably best in the ZIP to say that expiry height does not apply to coinbase transactions, rather than saying the value must be zero, as it then gives us a field we can re-use in a future ZIP e.g. stashing a merkle root in the expiry height field of a coinbase transaction.

Reply:
I don't particularly like the idea of using a field for two different purposes (as opposed to using a field that was previously reserved), but that's an orthogonal issue. Ignoring it for coinbase is fine.